hmac.go

```diff
@@ -41,19 +41,26 @@ func (h hmacer) verifySignature(signature, payload string) (bool, error) {
 	return hmac.Equal([]byte(expectedSignature), []byte(signature)), nil
 }
 
-func (h hmacer) parseSignature(signatureKeyPair string) (string, error) {
-	if !strings.Contains(signatureKeyPair, "|") {
-		return "", SignatureError{"Signature-key pair does not contain |"}
+func (h hmacer) getMatchingSignature(signaturePairs string) (string, string) {
+	pairs := strings.Split(signaturePairs, "&")
+	for _, pair := range pairs {
+		split := strings.Split(pair, "|")
+		if len(split) == 2 && split[0] == h.publicKey {
+			return split[0], split[1]
+		}
 	}
-	split := strings.Split(signatureKeyPair, "|")
-	if len(split) != 2 {
-		return "", SignatureError{"Signature-key pair contains more than one |"}
+	return "", ""
+}
+
+func (h hmacer) parseSignature(signatureKeyPairs string) (string, error) {
+	if !strings.Contains(signatureKeyPairs, "|") {
+		return "", SignatureError{"Signature-key pair does not contain |"}
```

Comment:
This validation is probably out of date I think. Taking a look at the [Braintree Java SDKs implementation of this](https://github.com/braintree/braintree_java/blob/8bde15986e9c94ce6611cdb3c7f0776f795df5a4/src/main/java/com/braintreegateway/WebhookNotificationGateway.java#L22-L45) it doesn't look like it does this check. Instead it validates that only valid characters are included and then iterates through ignoring entries that don't have the pipe.